src/script/components/UserList/components/UserListItem/UserListItem.tsx

```diff
@@ -26,149 +26,115 @@ import {Checkbox, CheckboxLabel} from '@wireapp/react-ui-kit';
+  onKeyDown: (user: User, event: KeyboardEvent) => void;
+  user: User;
+  selfInTeam: boolean;
+  showArrow: boolean;
 }
 
-const ParticipantItem = <UserType extends User | ServiceEntity>(
-  props: React.PropsWithChildren<ParticipantItemProps<UserType>>,
-): React.ReactElement | null => {
-  const {
-    badge,
-    callParticipant,
-    canSelect,
-    customInfo,
-    external,
-    hideInfo,
-    highlighted = false,
-    isSelected,
-    isSelfVerified = false,
-    mode = UserlistMode.DEFAULT,
-    noInteraction = false,
-    noUnderline = false,
-    participant,
-    selfInTeam,
-    showDropdown = false,
-    onContextMenu = noop,
-    onClick = noop,
-    onKeyDown = noop,
-  } = props;
+const UserListItem = ({
+  badge,
+  canSelect,
+  customInfo,
+  external,
+  hideInfo,
+  isSelected,
+  isSelfVerified = false,
+  mode = UserlistMode.DEFAULT,
+  noInteraction,
+  noUnderline = false,
+  user,
+  selfInTeam,
+  onClick,
+  onKeyDown,
+}: UserListItemProps) => {
   const checkboxId = useId();
-
   const [isInViewport, setIsInViewport] = useState(false);
-  const isUser = participant instanceof User && !participant.isService;
-  const isService = participant instanceof ServiceEntity || participant.isService;
-  const isSelf = !!(participant as User).isMe;
-  const selfString = `(${capitalizeFirstChar(t('conversationYouNominative'))})`;
-  const hasCustomInfo = !!customInfo;
-  const isTemporaryGuest = isUser && (participant as User).isTemporaryGuest();
-  const hasUsernameInfo = isUser && !hideInfo && !hasCustomInfo && !isTemporaryGuest;
-  const isOthersMode = mode === UserlistMode.OTHERS;
 
   const {
     is_verified: isVerified,
     isDirectGuest,
     availability,
     expirationText,
-    name: participantName,
-  } = useKoSubscribableChildren(
-    // We need to make TS believe that this is a User, otherwise it will complain about
-    // the fields in the array that don't exist on ServiceEntity
-    participant as User,
-    participant instanceof User ? ['isDirectGuest', 'is_verified', 'availability', 'expirationText', 'name'] : ['name'],
-  );
+    name: userName,
+  } = useKoSubscribableChildren(user, ['isDirectGuest', 'is_verified', 'availability', 'expirationText', 'name']);
 
-  const isFederated = participant instanceof User && participant.isFederated;
+  const {isMe: isSelf, isFederated} = user;
+  const isTemporaryGuest = user.isTemporaryGuest();
+
+  const hasCustomInfo = !!customInfo;
+  const hasUsernameInfo = !hideInfo && !hasCustomInfo && !isTemporaryGuest;
+  const isOthersMode = mode === UserlistMode.OTHERS;
 
-  const {sharesCamera, sharesScreen, isActivelySpeaking, isMuted} = useKoSubscribableChildren(callParticipant, [
-    'sharesCamera',
-    'sharesScreen',
-    'isActivelySpeaking',
-    'isMuted',
-  ]);
+  const selfString = `(${capitalizeFirstChar(t('conversationYouNominative'))})`;
 
-  const contentInfoText = (() => {
+  const contentInfoText = useMemo(() => {
```

Comment:
Is `useMemo` needed here? They are not too heavy operations and we basically have all the values in deps array, so it will be called every time any of those values change anyway 🤔